ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/numberinput/v1/numberinput/clientlibs/site/js/numberinputview.js

```diff
@@ -79,15 +79,23 @@
             // only initialize if patterns are set
             if (this._model._jsonModel.editFormat || this._model._jsonModel.displayFormat || FormView.Utils.isUserAgent('safari')) {
                 if (this.widgetObject == null) {
-                    this.widgetObject = new NumericInputWidget(this.getWidget(), this._model)
+                    this.widgetObject = new NumericInputWidget(this.getWidget(), this._model, this)
```

Comment:
I would suggest return the input widget from the NumericInputWidget and addEventListeners there 

for example 
```
this.widgetObject =  new NumericInputWidget(this.getWidget(), this._model)
this.widgetObject.inputElement.addEventListener('blur', setActive)
this.widgetObject.inputElement.addEventListener('focus', setActive)
```

The other option would be to take focus and blur handlers as options in the NumericInputWidget

Reply:
We anyways don't have any contract defined between view and widget.